### PR TITLE
496 code block update

### LINKED ISSUE #496: Update code patterns
# 1. Bug template

## Process

1. Navigate to base > code
2. Navigate to patterns > code
3. Navigate to patterns > code snippet

## Expected result

- The design of these code blocks should match the ones in Juju (see images below).
- The different variants are: `<code>` element, `<pre><code>` element (code block), code block numbered, code snippet dark, code snippet light

## Current result

- The style is not the most up to date. 
- There shouldn't be a code-inverse, rather code-snippet-dark and code-snippet-light variants
- The title of code in patterns should be code numbered

## Screenshot

https://github.com/ubuntudesign/vanilla-design/tree/master/Code

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

* updated all code variants
* drive-by: updated color documentation
* drive-by: moved reset up in the main vanilla.scss

## QA

1. go to vf.demo.haus and see that all the code resembles the images
 * &lt;code> element - /base/code/ - [inline](https://github.com/ubuntudesign/vanilla-design/blob/master/Code/%3Ccode%3E.png)
 * &lt;pre>&lt;code> element  - /base/code/ - [dark code block](https://github.com/ubuntudesign/vanilla-design/blob/master/Code/Code%20block.png)
 * code snippets- /patterns/code-snippet/ - [code snippet dark](https://github.com/ubuntudesign/vanilla-design/blob/master/Code/Code%20snippet%20dark.png) and [code snippet]https://github.com/ubuntudesign/vanilla-design/blob/master/Code/Code%20snippet%20light.png
 * numbered code blocks - /patterns/code-numbered/ - [light](https://github.com/ubuntudesign/vanilla-design/blob/master/Code/Code%20block%20numbered.png)

## Details

Fixes #496

### CODE REVIEW

File: docs/settings/_color-settings.md [hunk 1]
```diff
@@ -3,18 +3,63 @@ collection: settings
...
+<td>$color-light</td>
+<td style="background-color: #fff;">#fff</td>
+</tr>
+<tr>
+<td>$color-dark</td>
+<td style="background-color: #000;">#000</td>
+</tr>
+<tr>
+<td>$color-negative</td>
+<td style="background-color: #df382c;;">#df382c </td>
```
Comment (barrymcgee) [inline docs/settings/_color-settings.md]: Erroneous semi-colon here

File: docs/settings/_color-settings.md [hunk 2]
```diff
@@ -3,18 +3,63 @@ collection: settings
...
+</tr>
+<tr>
+<td>$color-warning</td>
+<td style="background-color: #eca918;;">#eca918 </td>
```
Comment (barrymcgee) [inline docs/settings/_color-settings.md]: Another erroneous semi-colon here

**Comment by pmahnke** [comment]
@barrymcgee fixed double semi-colons

File: docs/patterns/_code-snippet.md [hunk 3]
```diff
@@ -19,18 +19,18 @@ Code snippet should be used when presenting the user with a small snippet of cod
     <button class="code-snippet__action">Copy to clipboard</button>
 </div>
 
-## .code-snippet--inverse
+## .code-snippet--light
```
Comment (barrymcgee) [inline docs/patterns/_code-snippet.md]: Should this not be `## .code-snippet--dark`

**Pull request merged**